app/Actions/Photo/Strategies/AddStandaloneStrategy.php

```diff
@@ -112,152 +177,106 @@ public function do(): Photo
-			$tmpFile = new TemporaryLocalFile($this->sourceFile->getExtension(), $this->sourceFile->getBasename());
-			$tmpFile->write($this->sourceFile->read());
-			$this->sourceFile->close();
-			$this->sourceFile = $tmpFile;
-			// Reset source file info to the new temporary and ensure that
-			// it will be deleted later
-			$this->parameters->importMode->setDeleteImported(true);
-		}
-
-		$absolutePath = $this->sourceFile->getAbsolutePath();
-		// If we are importing via symlink, we don't actually overwrite
-		// the source, but we still need to fix the dimensions.
-		$this->imageHandler->autoRotate(
-			$absolutePath,
-			$orientation,
-			$this->parameters->importMode->shallImportViaSymlink()
-		);
-
-		// stat info (filesize, access mode, etc.) are cached by PHP to avoid
-		// costly I/O calls.
-		// If cache is not cleared, the size before rotation is used and later
-		// yields an incorrect value.
-		clearstatcache(true, $absolutePath);
-	}
-
-	/**
 	 * @throws MediaFileOperationException
+	 * @throws ConfigurationException
 	 */
-	protected function handleGoogleMotionPicture(): void
+	private function putSourceIntoFinalDestination(FlysystemFile $targetFile): StreamStat
 	{
-		if ($this->parameters->exifInfo->microVideoOffset === 0) {
-			return;
-		}
-
-		$videoLengthBytes = $this->parameters->exifInfo->microVideoOffset;
-		$original = $this->photo->size_variants->getOriginal();
-		$shortPathPhoto = $original->short_path;
-		$fullPathPhoto = $original->full_path;
-		$shortPathVideo = pathinfo($shortPathPhoto, PATHINFO_FILENAME) . '.mov';
-		$fullPathVideo = pathinfo($fullPathPhoto, PATHINFO_FILENAME) . '.mov';
-
 		try {
-			// 1. Extract the video part
-			$fp = fopen($fullPathPhoto, 'r');
-			$fp_video = tmpfile(); // use a temporary file, will be deleted once closed
-
-			// The MP4 file is located in the last bytes of the file
-			fseek($fp, -1 * $videoLengthBytes, SEEK_END); // It needs to be negative
-			$data = fread($fp, $videoLengthBytes);
-			fwrite($fp_video, $data, $videoLengthBytes);
+			if ($this->parameters->importMode->shallImportViaSymlink()) {
+				if (!$targetFile->isLocalFile()) {
+					throw new ConfigurationException('Symlinking is only supported on local filesystems');
+				}
+				$targetPath = $targetFile->toLocalFile()->getPath();
+				$sourcePath = $this->sourceFile->getRealPath();
+				// For symlinks we must manually create a non-existing
+				// parent directory.
+				// This mimics the behaviour of Flysystem for regular files.
+				$targetDirectory = pathinfo($targetPath, PATHINFO_DIRNAME);
+				if (!is_dir($targetDirectory)) {
+					\Safe\mkdir($targetDirectory, 0777, true);
+				}
+				\Safe\symlink($sourcePath, $targetPath);
+				$streamStat = StreamStat::createFromLocalFile($this->sourceFile);
+			} else {
+				// Nothing to do for non-JPEGs or correctly oriented photos.
+				// TODO: Why do we only normalize JPEG? Should it be sufficient that we have a successfully loaded source image?
```

Comment:
My guess is that it's for historical reasons: until we added support for exiftool, exif info was only available for jpeg files. So we can probably remove the test for `image/jpeg` below. There's a set of samples in different formats to try at https://zpl.fi/exif-orientation-in-different-formats/

Reply:
Fixed in https://github.com/LycheeOrg/Lychee/pull/1351/commits/11543f4c5c16b58eb5b4bd999bd6ea3f5f58fb65.

Our test suite now contains images which requires normalization such that this code path is actually tested. But when I added the photos, I noticed that today's browser seem to interpret EXIF data and do an auto-rotation in the browser, if necessary. I tested it with Firefox and Chrome on Linux and with the Google browser and Firefox Mobile on my Android phone. So may be the whole auto-rotation thing is not necessary anymore  after all.

Here are the sample files:

 - https://github.com/LycheeOrg/Lychee/blob/master/tests/Samples/orientation-90.jpg
 - https://github.com/LycheeOrg/Lychee/blob/master/tests/Samples/orientation-180.jpg
 - https://github.com/LycheeOrg/Lychee/blob/master/tests/Samples/orientation-270.jpg
 - https://github.com/LycheeOrg/Lychee/blob/master/tests/Samples/orientation-hflip.jpg
 - https://github.com/LycheeOrg/Lychee/blob/master/tests/Samples/orientation-vflip.jpg

But I would still keep it, because it has always been there.

Reply:
Yeah, I have wondered whether we should be rotating the original images at all, whether during import or during explicit rotation. However, for images that lack the `medium` variants (due to their low resolution), we use the `big` one instead, so we would need to add support for on-the-fly rotation to the front end.

Reply:
But that's independent of this PR...